File: Level 3/queue-to-do/solution.py
import operator

def solution(start, length):
    checksum = 0

    for row, firstID in enumerate(xrange(start, start + length ** 2, length)):               # Creates enumerated "rows" with all of the numbers.
        checksum ^= reduce(operator.xor, xrange(firstID, firstID + length - row))            # Cumulitavely XORs the "rows" using reduce, ignoring the necessary digits at the end.
    
    return checksum

File: Level 3/queue-to-do/Solution.java
class Solution {
    public static int solution(int start, int length) {
        int checksum = 0;

        for (int row = 0; row < length; row ++, start += length) {   // Iterates through every row, shifting the start by the length each time.
            for (int i = start; i < start + length - row; i++) {     // Iterates through every number in the row, ignoring "row" numbers at the end.
                checksum ^= i;
            }
        }
        
        return checksum;
    }
}